RecoLocalTracker/SiStripZeroSuppression/src/SiStripAPVRestorer.cc

```diff
@@ -976,79 +942,280 @@ bool SiStripAPVRestorer::CheckBaseline(const std::vector<int16_t> &baseline) con
+  for(int16_t strip = (APVn-firstAPV)*128; strip < (APVn-firstAPV+1)*128; ++strip) singleAPVdigi.push_back(digis[strip]+1024);
+
+  digimap_t discontinuities;   // it will contain the start and the end of each region in which a greadient is present
+  discontinuities.clear();
+
+  digimap_t::iterator itdiscontinuities;
+
+  //----Variables of the first part---//
+
+  bool isMinimumAndNoMax = false;
+  bool isFirstStrip = false;
+
+  //---Variables of the second part---//
+
+  int actualStripADC = 0;
+  int previousStripADC = 0;
+
+
+  int greadient = 0;
+  int maximum_value=0;
+  int high_maximun_cluster = 1025 + 1024;
+  int number_good_minimum = 0;
+  int first_gradient = 0;
+  int strip_first_gradient = 0;
+  int ADC_start_point_cluster_pw = 0;
+  int auxiliary_end_cluster = 0;
+  int first_start_cluster_strip = 0;
+  int first_start_cluster_ADC = 0;
+  bool isAuxiliary_Minimum = false;
+  bool isPossible_wrong_minimum = false;
+  bool isMax =false;
+
+  //----------SECOND PART: CLUSTER FINDING--------//
+
+  for(uint16_t strip=0; strip < singleAPVdigi.size(); ++strip){
+    if (strip == 0) {
+      actualStripADC = singleAPVdigi[strip];
+      if (abs(singleAPVdigi[strip]-singleAPVdigi[strip+1])>gradient_threshold_){
+        isFirstStrip=true;
+        isMinimumAndNoMax=true;
+        discontinuities.insert(discontinuities.end(), std::pair<int, int >(strip, actualStripADC));
+      } else if (actualStripADC > (gradient_threshold_+1024)) {
+          discontinuities.insert(discontinuities.end(), std::pair<uint16_t, int16_t >(strip, 0+1024));
+          isMinimumAndNoMax=true;
+          first_start_cluster_strip=strip;
+          first_start_cluster_ADC=1024;
+      }
+
+    } else if (strip>0) {
+      previousStripADC = actualStripADC;
+      actualStripADC = singleAPVdigi[strip];
+      greadient = actualStripADC - previousStripADC;
+
+      if (((greadient> gradient_threshold_)&&(isMax==false)&&(isMinimumAndNoMax==false))){
+        isMax=false;
+        if (((abs(maximum_value - previousStripADC) < (2*gradient_threshold_))&&discontinuities.size()>1)){ // agregar el || del tamano de cluster                                      //para verificar que el ruido no interfiera y se detecten falsos hits
+                isPossible_wrong_minimum=true;
+                itdiscontinuities=discontinuities.end();
+                --itdiscontinuities;
+                if(discontinuities.size()>1){
+                        --itdiscontinuities;
+                }
+                strip_first_gradient= itdiscontinuities->first;
+                ADC_start_point_cluster_pw= itdiscontinuities->second;
+                first_gradient = abs(ADC_start_point_cluster_pw - singleAPVdigi[strip_first_gradient+1]);
+                ++itdiscontinuities;
+                discontinuities.erase(itdiscontinuities);
+                itdiscontinuities=discontinuities.end();
+                --itdiscontinuities;
+                discontinuities.erase(itdiscontinuities);
```

Comment:
```
discontinuities.erase(++itdiscontinuities);
discontinuities.erase(--discontinuities.end());
```
should be able to replace the 5 lines above.
Similar pattern(s) apply to more places below